ocs_ci/utility/kms.py

```diff
@@ -0,0 +1,503 @@
+        """
+        if config.ENV_DATA.get("VAULT_BACKEND_PATH"):
+            self.vault_backend_path = config.ENV_DATA.get("VAULT_BACKEND_PATH")
+        else:
+            # Generate backend path name using prefix "ocs"
+            # "ocs-<cluster-id>"
+            self.cluster_id = self.get_cluster_id()
+            self.vault_backend_path = (
+                f"{constants.VAULT_DEFAULT_PATH_PREFIX}-{self.cluster_id}"
+            )
+        cmd = f"vault secrets enable -path={self.vault_backend_path} kv"
+        out = subprocess.check_output(shlex.split(cmd))
+        if "Success" in out.decode():
+            logger.info(f"vault path {self.vault_backend_path} created")
+        else:
+            raise VaultPathCreationFailed(
+                f"Failed to create path f{self.vault_backend_path}"
+            )
+        self.vault_create_policy()
+
+    def vault_create_policy(self):
+        """
+        Create a vault policy and generate token
+
+        """
+        policy = (
+            f'path "{self.vault_backend_path}/*" {{\n'
+            f'  capabilities = ["create", "read", "update","delete"]'
+            f"\n}}\n"
+            f'path "sys/mounts" {{\n'
+            f'capabilities = ["read"]\n'
+            f"}}"
+        )
+        vault_hcl = tempfile.NamedTemporaryFile(mode="w+", prefix="test", delete=False)
+        with open(vault_hcl.name, "w") as hcl:
+            hcl.write(policy)
+
+        if not config.ENV_DATA.get("VAULT_POLICY"):
+            self.vault_policy_name = (
+                f"{constants.VAULT_DEFAULT_POLICY_PREFIX}-" f"{self.cluster_id}"
+            )
+        else:
+            self.vault_policy_name = config.ENV_DATA.get("VAULT_POLICY")
+
+        cmd = f"vault policy write {self.vault_policy_name} {vault_hcl.name}"
+        out = subprocess.check_output(shlex.split(cmd))
+        if "Success" in out.decode():
+            logger.info(f"vault policy {self.vault_policy_name} created")
+        else:
+            raise VaultPolicyCreationFailed(
+                f"Failed to create policy f{self.vault_policy_name}"
+            )
+        self.vault_path_token = self.generate_vault_token()
+
+    def generate_vault_token(self):
+        """
+        Generate a token for self.vault_policy_name
+
+        Returns:
+            str: vault token
+
+        """
+        cmd = f"vault token create -policy={self.vault_policy_name} " f"--format=json"
+        out = subprocess.check_output(shlex.split(cmd))
+        json_out = json.loads(out)
+        return json_out["auth"]["client_token"]
+
+    def get_cluster_id(self):
+        """
+        Get cluster UUID
+        Not relying on metadata.json as user sometimes want to run
+        only with kubeconfig for some tests
+
+        Returns:
+            str: cluster UUID
+
+        """
+        cluster_id = run_cmd(
+            "oc get clusterversion version -o jsonpath='{.spec.clusterID}'"
+        )
+        return cluster_id
```

Comment:
get_cluster_id is generic one. we can move this function out of this module

Reply:
@vavuthu https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/utility/utils.py#L3042 already there exists a function with the same name but its using metadata.json which could be problematic at times (when user wants to run test only with kubeconfig) hence I added this function locally. Do you have any other location in mind where I can put this function ?

Reply:
may be we can change function name to "get_running_cluster_id" and add doc string that Get the cluster from the running cluster

Reply:
ack

Reply:
done